components/script/dom/document.rs

```diff
@@ -1343,6 +1365,9 @@ impl Document {
         event.fire(target);
         let mut cancel_state = event.get_cancel_state();
 
+        self.handle_event(event);
```

Comment:
I think we should inline the code from handle_event in here instead. No need to call a different method.

Reply:
@jdm If we move the code from `handle_event` here, I think we don't need `impl VirtualMethods for Document`?

Reply:
Go ahead and remove it if that's the case.

Reply:
Thanks! I'll try it!

Reply:
According to [this comment](https://github.com/servo/servo/pull/13460/files#r83593226) in the most original PR from @nox , I think that's why nox made it inside `VirtualMethods`.
However, after trying to make the code in here, the focus sequential still works.
Not sure if there's any concerns about moving the code in here instead of inside `VirtualMethods`?

Reply:
The comment you point out was made in a version that did not include the parent type's handle_event call, which is required when using VirtualMethods. I don't think we need to use VirtualMethods for that code, however.

Reply:
Understood. We can handle the event without `VirtualMethods` but I didn't know too much about it; thus, I left the comment.
Now, I think it should be okay to remove it :)